packages/volto/src/components/manage/Widgets/IdWidget.jsx

```diff
@@ -46,8 +46,8 @@ const IdWidget = (props) => {
   const intl = useIntl();
   const dispatch = useDispatch();
   const ref = useRef();
-
-  const indexes = useSelector((state) => keys(state.querystring.indexes));
+  const indexesObj = useSelector((state) => state.querystring.indexes);
+  const indexes = useMemo(() => keys(indexesObj), [indexesObj]);
```

Comment:
... also another remark... it would be better to rename `values` to `value`, as it's clearly a single string and not an array of strings, so the plural is not correct name imo.

Reply:
done as you said.

Maybe I am wrong. But useSelector may re-render the component if you provide the new object every time. 

https://github.com/reduxjs/react-redux/issues/1400#issuecomment-534181012